sys/progress_bar/progress_bar.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_progress_bar
+ * @{
+ *
+ * @file
+ * @brief       Progress bar implementation
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#include "progress_bar.h"
+
+void progress_bar_update(progress_bar_t *progress_bar)
+{
+    assert(progress_bar->value < 101);
+
+    /* Hide cursor */
+    printf("\033[?25l");
+
+    /* Move cursor at the beginning of the line */
+    putchar('\r');
+
+    /* Display progress bar prefix if any */
+    if (progress_bar->prefix[0]) {
+        printf("%s", progress_bar->prefix);
+    }
+
+    printf(PROGRESS_BAR_PREFIX_CHARACTER);
+
+    /* Fully reprint the progress bar */
+    for (unsigned j = 0; j < PROGRESS_BAR_LENGTH; j++) {
+        if (100 * j < (uint16_t)(progress_bar->value * PROGRESS_BAR_LENGTH)) {
```

Comment:
If you change this to use 0-255 instead of 0-100, the two multiplications turn into shifts. Also, the assert on top of this function becomes obsolete. Similar at the call site.

Not sure that justifies not using percentages.